.github/actions/issue-validator/src/index.mjs

```diff
@@ -26,146 +24,61 @@ async function run() {
     const { payload, repo } = github.context
     const { issue, pull_request } = payload
 
-    if (pull_request || !issue?.body) return
+    if (pull_request || !issue?.body || !process.env.GITHUB_TOKEN) return
 
     /** @type {Label} */
     const newLabel = payload.label
     const { body, number: issueNumber } = issue
-    /** @type {Label[]} */
-    const labels = issue.labels
-
-    if (debug) {
-      core.info(
-        `Validating issue ${issueNumber}:
-    Labels:
-      New: ${json(newLabel)}
-      All: ${json(labels)}
-    Body: ${body}`
-      )
-    }
-
-    const isBugReport = newLabel.name === bugReportLabel
-
-    const isManuallyLabeled = labels.some(
-      (label) => label.name === verifyCanaryLabel
-    )
-
-    if (!isBugReport && !isManuallyLabeled) {
-      return core.info(
-        'Issue is ignored, because it is not a bug report or is not manually labeled'
-      )
-    }
-
-    if (!process.env.GITHUB_TOKEN) {
-      throw new Error('GITHUB_TOKEN is not set')
-    }
-
     const client = github.getOctokit(process.env.GITHUB_TOKEN).rest
-
-    /**
-     * @param {string} label
-     * @param {string} comment
-     */
-    function notifyOnIssue(label, comment) {
-      const issueCommon = { ...repo, issue_number: issueNumber }
-
-      if (debug) {
-        core.info('Skipping comment/label because we are in DEBUG mode')
-        core.info(json({ label, comment }))
-        return
+    const issueCommon = { ...repo, issue_number: issueNumber }
+
+    /** @param {string|null|undefined} link */
+    async function hasRepro(link) {
+      if (!link) return false
+      try {
+        new URL(link)
+      } catch {
+        return false
       }
-
-      return Promise.all([
-        client.issues.addLabels({ ...issueCommon, labels: [label] }),
-        client.issues.createComment({ ...issueCommon, body: comment }),
-      ])
+      if (['example.com'].includes(link)) return false
+      const response = await fetch(link)
+      if (!response.ok) return false
     }
 
-    const isVerifyCanaryChecked = body.includes(
-      '- [X] I verified that the issue exists in Next.js canary release'
+    const hasValidRepro = await hasRepro(
+      body
+        .toLowerCase()
```

Comment:
I think you need to remove this "toLowerCase()` because it will never match the next line of "To Reproduce"